qiskit/transpiler/passes/routing/swap_strategies/swap_strategy.py

```diff
@@ -0,0 +1,305 @@
+    are swapped. At each swap layer SWAP gates are applied to the corresponding edges. This class
+    stores the permutations of the qubits resulting from the swap strategy. Swap strategies are
+    intended to be used on blocks of commuting gates which are often present in variational
+    algorithms.
+    """
+
+    def __init__(self, coupling_map: CouplingMap, swap_layers: List[List[Tuple[int, int]]]) -> None:
+        """
+        Args:
+            coupling_map: The coupling map the strategy is implemented for.
+            swap_layers: The swap layers of the strategy, specified as a list of sets of
+                edges (edges can be represented as lists, sets or tuples containing two integers).
+
+        Raises:
+            QiskitError: If the coupling map is not specified.
+            QiskitError: if the swap strategy is not valid. A swap strategy is valid if all
+                swap gates, specified as tuples, are contained in the edge set of the coupling map.
+        """
+        self._coupling_map = coupling_map
+        self._num_vertices = coupling_map.size()
+        self._swap_layers = swap_layers
+        self._distance_matrix = None
+        self._possible_edges = None
+        self._inverse_composed_permutation = {0: list(range(self._num_vertices))}
+
+        edge_set = set(self._coupling_map.get_edges())
+
+        for i, layer in enumerate(self._swap_layers):
+            for edge in layer:
+                if edge not in edge_set:
+                    raise QiskitError(
+                        f"The {i}th SWAP layer contains the edge {edge} which is not "
+                        f"part of the underlying coupling map with {edge_set} edges."
+                    )
+
+    def __len__(self) -> int:
+        """Return the length of the strategy as the number of layers.
+
+        Returns:
+            The number of layers of the swap strategy.
+        """
+        return len(self._swap_layers)
+
+    def __repr__(self) -> str:
+        """Representation of the swap strategy.
+
+        Returns:
+            The represenation of the swap strategy.
+        """
+        description = [f"{self.__class__.__name__} with swap layers:\n"]
+
+        for layer in self._swap_layers:
+            description.append(f"{layer},\n")
+
+        description.append(f"on {self._coupling_map} coupling map.")
+        description = "".join(description)
+
+        return description
+
+    def swap_layer(self, idx: int) -> List[Tuple[int, int]]:
+        """Return the layer of swaps at the given index.
+
+        Args:
+            idx: The index of the returned swap layer.
+
+        Returns:
+            A copy of the swap layer at ``idx`` to avoid any unintentional modification to
+            the swap strategy.
+        """
+        return list(swap for swap in self._swap_layers[idx])
+
+    @property
+    def distance_matrix(self) -> List[List[int]]:
```

Comment:
It is different. Here, the distance matrix is the number of swap layers that need to be applied before two qubits become adjacent. The distance_matrix in the coupling map looks like a physical distance between qubits which is not the same as the distance matrix in the swap strategy even when using the `swapped_coupling_map`. You can further see the difference because `swapped_coupling_map` is in fact a method that takes an argument, namely the index of the swap layer whereas `SwapStrategy.distance_matrix` is swap-layer-index independent.